### PR TITLE
Core: Remove TableOperations from metadata tables.

### PR DESCRIPTION
**Pull request by rdblue**
This removes the use of `TableOperations` from metadata tables. Where `TableMetadata` is needed, operations is fetched from the `Table` instance instead of passing the ops explicitly. This is in preparation for metadata tables that are not tied to a particular Iceberg table.

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/BaseTable.java [hunk 1]
```diff
@@ -229,17 +229,17 @@ public Transaction newTransaction() {
 
   @Override
   public FileIO io() {
-    return operations().io();
+    return ops.io();
```
Comment (szehon-ho) [inline core/src/main/java/org/apache/iceberg/BaseTable.java:232]: These are unrelated , and just simplification/standardization, right?  Fine with me either way

**Review by szehon-ho — APPROVED** [review]
Looks good to me.  I guess it also as a consequence removes TableOperations from the BaseTableScan hierarchy, but seems to me a good simplification if we never needed it there.

**Comment by nastra** [comment]
@rdblue looks like this needs some RevAPI changes because the APIs changed

**Comment by rdblue** [comment]
Thanks for the review, @szehon-ho!

File: core/src/main/java/org/apache/iceberg/BaseTable.java (hunk 1 above)
Comment (rdblue) [reply core/src/main/java/org/apache/iceberg/BaseTable.java:232]: Where possible, I removed references to methods that expose `TableOperations` and deprecated the methods.

Here in `BaseTable`, I just wanted to be more consistent and use the instance field directly.
